pkg/utils/kubernetes/health/monitoring_test.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2024 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package health_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	"github.com/gardener/gardener/pkg/utils/kubernetes/health"
+)
+
+var _ = Describe("Monitoring", func() {
+	DescribeTable("CheckPrometheus",
+		func(prometheus *monitoringv1.Prometheus, matcher types.GomegaMatcher) {
+			err := health.CheckPrometheus(prometheus)
+			Expect(err).To(matcher)
+		},
+		Entry("healthy", &monitoringv1.Prometheus{
+			Spec:   monitoringv1.PrometheusSpec{CommonPrometheusFields: monitoringv1.CommonPrometheusFields{Replicas: ptr.To(int32(1))}},
+			Status: monitoringv1.PrometheusStatus{Replicas: 1, AvailableReplicas: 1, Conditions: []monitoringv1.Condition{{Type: monitoringv1.Available, Status: monitoringv1.ConditionTrue}}},
+		}, BeNil()),
+		Entry("healthy with nil replicas", &monitoringv1.Prometheus{
+			Status: monitoringv1.PrometheusStatus{Replicas: 1, AvailableReplicas: 1, Conditions: []monitoringv1.Condition{{Type: monitoringv1.Available, Status: monitoringv1.ConditionTrue}}},
+		}, BeNil()),
+		Entry("not observed at latest version", &monitoringv1.Prometheus{
+			ObjectMeta: metav1.ObjectMeta{Generation: 1},
+			Status:     monitoringv1.PrometheusStatus{Conditions: []monitoringv1.Condition{{Type: monitoringv1.Available, Status: monitoringv1.ConditionTrue}}},
+		}, HaveOccurred()),
```

Comment:
Shouldn't we test on the error string ?

```suggestion
		}, MatchError(ContainSubstring("observed generation outdated (0/1)"))),
```

Reply:
Sure, fair enough, I copied this from older tests which use `HaveOccurred` only. Anyways, implemented, thanks :)